blocks/author-profile/author-profile.js

```diff
@@ -26,34 +26,49 @@ function extractAuthorDescription(details) {
   return description;
 }
 
-async function getAuthorDetails() {
-  // #todo: handle multiple authors
-  const name = getMetadata('author');
-  const result = name
-    ? await ffetch('/authors-index.json').filter((entry) => entry.author === name).limit(1).all() : null;
-  const details = (!result || result.length < 1)
-    ? { path: '/author/name', author: 'Name' } : result[0];
-  details.description = extractAuthorDescription(details);
-  return details;
+function completeEntry(entry) {
+  const result = (!entry) ? { path: '/author/name', author: 'Name' } : entry;
+  result.description = extractAuthorDescription(result);
+  return result;
 }
 
-export default async function decorate(block) {
-  const details = await getAuthorDetails();
-  const authorImage = details.image
-    ? createOptimizedPicture(details.image, details.author, false, breakpoints) : null;
+async function getAuthorEntry(entryFilter) {
+  const result = await ffetch('/authors-index.json').filter(entryFilter).limit(1).all();
+  return completeEntry((!result || result.length < 1) ? null : result[0]);
+}
+
+function renderProfile(entry) {
+  const authorImage = entry.image
+    ? createOptimizedPicture(entry.image, entry.author, false, breakpoints) : null;
   const authorProfile = div(
     div({ class: 'avatar' }, authorImage ? div(authorImage) : div()),
     div(
       { class: 'details' },
-      h2(details.author),
-      p(details.description),
+      h2(entry.author),
+      p(entry.description),
       p(
         { class: 'link' },
-        a({ href: details.path, 'aria-label': 'Read more' }, 'See more by this author'),
+        a({ href: entry.path, 'aria-label': 'Read more' }, 'See more by this author'),
         span({ class: 'icon icon-link-arrow' }),
       ),
     ),
   );
   decorateIcons(authorProfile);
+  return authorProfile;
+}
+
+export default async function decorate(block) {
+  // #todo: handle multiple authors
```

Comment:
Why not implementing this?

Reply:
There's a bunch of requirements for multiple authors on article pages, and the design is not yet ready. Will be covered in ticket https://github.com/urfuwo/hlx-test/issues/246